plugins/modules/web_infrastructure/jboss.py

```diff
@@ -142,7 +142,7 @@ def main():
             # Clean up old failed deployment
             os.remove(os.path.join(deploy_path, "%s.failed" % deployment))
 
-        shutil.copyfile(src, os.path.join(deploy_path, deployment))
+        shutil.copy(src, os.path.join(deploy_path, deployment))
```

Comment:
```suggestion
        module.preserved_copy(src, os.path.join(deploy_path, deployment))
```

Taking this a step further I would use the `Ansible` provided [utilities](https://github.com/ansible/ansible/blob/58b03be4178fda12a7d082ae9fd54104229e9d99/lib/ansible/module_utils/basic.py#L1623) if appropriate to preserve as much of the source file's attributes as possible.